src/zkevm_specs/evm/step.py

```diff
@@ -95,3 +96,25 @@ def __init__(
         self.src_addr_end = FQ(src_addr_end)
         self.from_tx = FQ(from_tx)
         self.tx_id = FQ(tx_id)
+
+
+class CopyCodeToMemoryAuxData:
+    src_addr: FQ
+    dst_addr: FQ
+    bytes_left: FQ
+    src_addr_end: FQ
+    code: Bytecode
```

Comment:
`aux_data` is tracked as witness in circuit, so we should just have `code_hash` here.
```suggestion
    code_hash: RLC
```

Reply:
I used the entire bytecode in `aux_data` because I wanted a way to get the `is_code` field of the bytecode table row.

Is there another way I can achieve that? I am referring to this part:
https://github.com/scroll-tech/zkevm-specs/blob/feat/codecopy/src/zkevm_specs/evm/execution/copy_code_to_memory.py#L20-L30

Reply:
Ah I see. You can make the `is_code` to be optional, then don't pass it should work. (note that `Instruciton` also needs to be updated to have `is_code: bool = None`)

https://github.com/appliedzkp/zkevm-specs/blob/96c10a2d188d913421cfcfe34c38f35d951d200b/src/zkevm_specs/evm/table.py#L364-L377

```diff
-         is_code: Expression,
+         is_code: Expression = None,
```

Reply:
I initially thought that it isn't required to check `is_code`, since it is simply copying byte-by-byte to memory. So I did consider having `is_code` optional. Still wanted to check with you, so that's cool, I will update this part of the codebase :)

Reply:
I see, currently in circuit we indeed need such extra auxiliary information to make lookup work. We can try to not include `is_code` column for such case in the future.

Reply:
Yes, as I can see, the `opcode_lookup` in fact needs the `is_code` field, whereas something like `CODECOPY` that copies byte-by-byte does not really care about `is_code`. Correct me if I'm wrong.